src/plopp/widgets/drawing.py

```diff
@@ -0,0 +1,142 @@
+
+        self._figure = figure
+        self._destination_is_fig = is_figure(self._figure)
+        self._input_node = input_node
+        self._draw_nodes = {}
+        self._output_nodes = {}
+        self._func = func
+        self._tool = tool(ax=self._figure.ax, autostart=False)
+        self._destination = destination
+        self._tool.on_create(self.make_node)
+        self._tool.on_change(self.update_node)
+        self._tool.on_remove(self.remove_node)
+
+    def convert_raw_data(self, artist):
+        return lambda: artist
+
+    def make_node(self, artist):
+        draw_node = Node(self.convert_raw_data(artist))
+        nodeid = draw_node.id
+        self._draw_nodes[nodeid] = draw_node
+        artist.nodeid = nodeid
+        output_node = node(self._func)(self._input_node, draw_node)
+        self._output_nodes[nodeid] = output_node
+        if self._destination_is_fig:
+            output_node.add_view(self._destination)
+            self._destination.update(new_values=output_node(), key=output_node.id)
+            self._destination.artists[output_node.id].color = artist.color
+        elif isinstance(self._destination, Node):
+            self._destination.parents.append(output_node)
+
+    def update_node(self, artist):
+        n = self._draw_nodes[artist.nodeid]
+        n.func = self.convert_raw_data(artist)
+        n.notify_children(artist)
+
+    def remove_node(self, artist):
+        nodeid = artist.nodeid
+        draw_node = self._draw_nodes[nodeid]
+        output_node = self._output_nodes[nodeid]
+        if self._destination_is_fig:
+            self._destination.artists[output_node.id].remove()
+        output_node.remove()
+        draw_node.remove()
+
+    def start_stop(self):
+        """
+        Toggle start or stop of the tool.
+        """
+        if self.value:
+            self._tool.start()
+        else:
+            self._tool.stop()
+
+
+class PointsTool(DrawingTool):
+    """
+    Tool to add point markers onto a figure.
+
+    Parameters
+    ----------
+    figure:
+        The figure where the tool will draw things (points, lines, shapes...).
+    input_node:
+        The node that provides the raw data which is shown in ``figure``.
+    tool:
+        The Mpltoolbox tool to use (Points, Lines, Rectangles, Ellipses...).
+    func:
+        The function to be used to make a node whose parents will be the ``input_node``
+        and a node yielding the current state of the tool (current position, size).
+    destination:
+        Where the output from the ``func`` node will be then sent on. This can either
+        be a figure, or another graph node.
+    value:
+        Activate the tool upon creation if ``True``.
+    **kwargs:
+        Additional arguments are forwarded to the ``ToggleTool`` constructor.
+    """
+
+    def __init__(self, **kwargs):
+        from mpltoolbox import Points
+        super().__init__(tool=partial(Points, mec='w'), icon='crosshairs', **kwargs)
```

Comment:
Is inheritance the right solution here? You do not seem to override a significant number of methods. Can we just inject the `tool` and `convert_raw_data` function when creating a `DrawingTool`, instead of inheriting?

Reply:
i'll try that and see how it works

Reply:
So would you:
1. make `DrawingTool` inherit from `ToggleTool` and inject the `Points` and `convert_raw_data`
2. convert `DrawingTool` to an `DrawEventHandler`, make `PointsTool` inherit from `ToggleTool` and give it a member `self.event_handler = DrawEventHandler()`
?

Reply:
Not sure, I didn't suggest to change `DrawingTool`. But if you look at the code now, `PointsTool` is almost just a "constructor" for `DrawingTool`. You could change it to a function that creates a specific `DrawingTool`.

The other things you mention we can consider later, maybe?